pandas/core/indexes/accessors.py

```diff
@@ -228,18 +237,20 @@ class PeriodProperties(Properties):
     """
 
 
-PeriodProperties._add_delegate_accessors(
-    delegate=PeriodIndex,
-    accessors=PeriodIndex._datetimelike_ops,
-    typ='property')
-PeriodProperties._add_delegate_accessors(
-    delegate=PeriodIndex,
-    accessors=PeriodIndex._datetimelike_methods,
-    typ='method')
-
-
 class CombinedDatetimelikeProperties(DatetimeProperties, TimedeltaProperties):
     # This class is never instantiated, and exists solely for the benefit of
     # the Series.dt class property. For Series objects, .dt will always be one
     # of the more specific classes above.
     __doc__ = DatetimeProperties.__doc__
+
+    @classmethod
+    def _make_accessor(cls, values):
+        try:
+            return maybe_to_datetimelike(values)
+        except Exception:
+            msg = "Can only use .dt accessor with datetimelike values"
+            raise AttributeError(msg)
+
+
+DatetimeAccessor = CombinedDatetimelikeProperties
+# Alias to mirror CategoricalAccessor
```

Comment:
If we're subclassing `PandasDelegate`, would it make more sense for these to be named e.g. `DatetimeDelegate`?